homeassistant/components/nuki/binary_sensor.py

```diff
@@ -0,0 +1,82 @@
+"""Doorsensor Support for the Nuki Lock."""
+
+import logging
+
+from pynuki import STATE_DOORSENSOR_OPENED
+
+from homeassistant.components.binary_sensor import DEVICE_CLASS_DOOR, BinarySensorEntity
+
+from . import NukiEntity
+from .const import (
+    ATTR_LOCK_DOORSENSOR_STATE,
+    ATTR_LOCK_DOORSENSOR_STATE_NAME,
+    ATTR_NUKI_ID,
+    DATA_COORDINATOR,
+    DATA_LOCKS,
+    DOMAIN as NUKI_DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Nuki lock binary sensor."""
+    data = hass.data[NUKI_DOMAIN][entry.entry_id]
+    coordinator = data[DATA_COORDINATOR]
+
+    entities = []
+
+    for lock in data[DATA_LOCKS]:
+        if lock.is_door_sensor_activated:
+            entities.extend([NukiDoorsensorEntity(coordinator, lock)])
+
+    async_add_entities(entities)
+
+
+class NukiDoorsensorEntity(NukiEntity, BinarySensorEntity):
+    """Representation of a Nuki Lock Doorsensor."""
+
+    @property
+    def name(self):
+        """Return the name of the lock."""
+        return self._nuki_device.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return str(self._nuki_device.nuki_id) + "-doorsensor"
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        data = {
+            ATTR_NUKI_ID: self._nuki_device.nuki_id,
+            ATTR_LOCK_DOORSENSOR_STATE: self._nuki_device.door_sensor_state,
+            ATTR_LOCK_DOORSENSOR_STATE_NAME: self._nuki_device.door_sensor_state_name,
+        }
+        return data
+
+    @property
+    def available(self):
+        """Return true if door sensor is present and activated."""
+        return self._nuki_device.is_door_sensor_activated
```

Comment:
Will this change if the connection goes down?

Reply:
When the doorsensor is disconnected from the lock this will return false.

Reply:
What happens if the connection to the lock goes down? Ie if we can't talk to the device api?

Reply:
When I disconnect the bridge from power I'm getting the following in the logs:
```
2021-03-02 17:43:32 ERROR (MainThread) [homeassistant.components.nuki] Timeout fetching nuki devices data
2021-03-02 17:43:32 DEBUG (MainThread) [homeassistant.components.nuki] Finished fetching nuki devices data in 10.001 seconds
``` 

But the entities in Homeassistant still remain available. Only if the bridge is not reachable on startup they will become unavailable (+ `Config entry 'XXX' for nuki integration not ready yet. Retrying in n seconds`)

Reply:
Yes, we need to use the CoordinatorEntity available method to check for availability besides the special case for door sensors.
```suggestion
        return super().available and self._nuki_device.is_door_sensor_activated
```

Reply:
I've implemented this for the binary_sensor & lock and it's working great!